demos/projects/ST/b-l475e-iot01a/config/core_http_config.h

```diff
@@ -0,0 +1,149 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_http_config_defaults.h
+ * @brief The default values for the configuration macros for the HTTP Client
+ * library.
+ *
+ * @note This file SHOULD NOT be modified. If custom values are needed for
+ * any configuration macro, a core_http_config.h file should be provided to
+ * the HTTP Client library to override the default values defined in this file.
+ * To use the custom config file, the HTTP_DO_NOT_USE_CUSTOM_CONFIG preprocessor
+ * macro SHOULD NOT be set.
+ */
+
+#ifndef CORE_HTTP_CONFIG_DEFAULTS_
+#define CORE_HTTP_CONFIG_DEFAULTS_
+
+/* *INDENT-OFF* */
+#ifdef __cplusplus
+    extern "C" {
+#endif
+/* *INDENT-ON* */
+
+/**
+ * @brief Maximum size, in bytes, of headers allowed from the server.
+ *
+ * If the total size in bytes of the headers received from the server exceeds
+ * this configuration, then the status code
+ * #HTTPSecurityAlertResponseHeadersSizeLimitExceeded is returned from
+ * #HTTPClient_Send.
+ *
+ * <b>Possible values:</b> Any positive 32 bit integer. <br>
+ * <b>Default value:</b> `2048`
+ */
+#ifndef HTTP_MAX_RESPONSE_HEADERS_SIZE_BYTES
+    #define HTTP_MAX_RESPONSE_HEADERS_SIZE_BYTES    2048U
+#endif
+
+/**
+ * @brief The HTTP header "User-Agent" value.
+ *
+ * The following header line is automatically written to
+ * #HTTPRequestHeaders_t.pBuffer:
+ * "User-Agent: my-platform-name\r\n"
+ *
+ * <b>Possible values:</b> Any string. <br>
+ * <b>Default value:</b> `my-platform-name`
+ */
+#ifndef HTTP_USER_AGENT_VALUE
+    #define HTTP_USER_AGENT_VALUE    "stml475-adu"
+#endif
+
+/**
+ * @brief The maximum duration between non-empty network reads while receiving
+ * an HTTP response via the #HTTPClient_Send API function.
+ *
+ * The transport receive function may be called multiple times until the end of
+ * the response is detected by the parser. This timeout represents the maximum
+ * duration that is allowed without any data reception from the network for the
+ * incoming response.
+ *
+ * If the timeout expires, the #HTTPClient_Send function will return
+ * #HTTPNetworkError.
+ *
+ * If #HTTPResponse_t.getTime is set to NULL, then this HTTP_RECV_RETRY_TIMEOUT_MS
+ * is unused. When this timeout is unused, #HTTPClient_Send will not retry the
+ * transport receive calls that return zero bytes read.
+ *
+ * <b>Possible values:</b> Any positive 32 bit integer. A small timeout value
+ * is recommended. <br>
+ * <b>Default value:</b> `10`
+ */
+#ifndef HTTP_RECV_RETRY_TIMEOUT_MS
+    #define HTTP_RECV_RETRY_TIMEOUT_MS    ( 10U )
```

Comment:
Please consider increasing the timeouts, especially for WiFi communication.

Reply:
Any suggestions on a better value?

Reply:
Maybe I misunderstood what these mean but I'd give it a try with 500/1000ms, especially since offline we were talking about some disconnects in certain cases.

Reply:
The issues I was running into were because my wifi hotspot is only connectable when my phone is open to the hotspot page for some reason 🙃 And the second issue was with disconnects during download, but that issue I think should be fixed with retries, not a longer timeout. So I'm not quite sure how to test whether a larger number is sufficient or not (especially since the comments for this say to keep it small)
@danewalton, did you have a way to repro the issues you were seeing?

Reply:
Discussed offline - will leave as 10U for now